scalameta/trees/src/main/scala/scala/meta/Trees.scala

```diff
@@ -373,7 +373,14 @@ object Defn {
                    templ: Template) extends Defn with Member.Type {
     // TODO: hardcoded in the @ast macro, find out a better way
     // require(templ.stats.getOrElse(Nil).forall(!_.is[Ctor]))
-    require(ctor.mods.isEmpty && ctor.paramss.isEmpty)
+
+    // TODO this doesn't work because the Dialect in implicit
```

Comment:
> For example, if you try to prettyprint such a tree in a Scala211 context, an exception must be thrown.

I agree, it seems we need to do the validation in some other place. However, I think scala.meta should **never** throw exceptions. I would prefer to have a `.validate(dialect)` method that I can call before `.syntax`. We can discuss different strategies later.